src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,395 @@
+                            GetNewExpirationTime(slidingExpiration.Value, utcNow));
+
+                        var recordsAffected = command.ExecuteNonQuery();
+                        //TODO: log if records not affected?
+                    }
+                }
+            }
+
+            StartScanForExpiredCacheItems();
+
+            return cacheItemvalue;
+        }
+
+        public async Task<byte[]> GetAsync([NotNull] string key)
+        {
+            byte[] cacheItemvalue = null;
+            var utcNow = _systemClock.UtcNow;
+
+            using (var connection = new SqlConnection(_options.ConnectionString))
+            {
+                var command = new SqlCommand(_sqlQueries.GetCacheItem, connection);
+                command.Parameters.AddWithValue(CacheItemId, key);
+                command.Parameters.AddWithValue("UtcNow", utcNow);
+
+                await connection.OpenAsync();
+
+                var reader = await command.ExecuteReaderAsync(CommandBehavior.SingleRow);
+
+                DateTimeOffset oldExpirationTimeUTC;
+                TimeSpan? slidingExpiration = null;
+                if (await reader.ReadAsync())
+                {
+                    // TODO: cache these oridnals as they are not going to change once the app starts
+                    var expiresAtTimeUTCIndex = reader.GetOrdinal(ExpiresAtTimeUTC);
+                    var cacheItemValueIndex = reader.GetOrdinal(CacheItemValue);
+                    var slidingExpirationInTicksIndex = reader.GetOrdinal(SlidingExpirationInTicks);
+
+                    oldExpirationTimeUTC = await reader.GetFieldValueAsync<DateTimeOffset>(expiresAtTimeUTCIndex);
+                    cacheItemvalue = await reader.GetFieldValueAsync<byte[]>(cacheItemValueIndex);
+
+                    if (!await reader.IsDBNullAsync(slidingExpirationInTicksIndex))
+                    {
+                        slidingExpiration = TimeSpan.FromTicks(
+                            await reader.GetFieldValueAsync<long>(slidingExpirationInTicksIndex));
+                    }
+                }
+                else
+                {
+                    return null;
+                }
+
+                // Check if the current item has a sliding expiration and is going to expire within the
+                // next timeout period(ex:20 minutes), then extend the expiration time.
+                if (slidingExpiration.HasValue)
+                {
+                    if (utcNow >= (oldExpirationTimeUTC - slidingExpiration.Value))
+                    {
+                        reader.Dispose();
+
+                        command = new SqlCommand(_sqlQueries.UpdateCacheItemExpiration, connection);
+                        command.Parameters.AddWithValue(CacheItemId, key);
+                        command.Parameters.AddWithValue(
+                            ExpiresAtTimeUTC,
+                            GetNewExpirationTime(slidingExpiration.Value, utcNow));
```

Comment:
If AbsoluteExpiration is set you need to account for that here. SlidingExpiration should not renew beyond AbsoluteExpiration.